### ISSUE TITLE
[Error] Parsing error on typed destructured object parameters

### ISSUE THREAD

**Issue by latourna** [eslint/typescript-eslint-parser#148]
**What version of TypeScript are you using?**
2.0.3

**What version of `typescript-eslint-parser` are you using?**
1.0.2

**What code were you trying to parse?**

```
foo({ bar, baz }: { bar: string, baz }) {
  //
}
```

**What did you expect to happen?**
Correct parsing

**What happened?**
```
0:0  error  Parsing error: Cannot read property 'loc' of null
```
Parsing fails when at least one parameter does not have a type assigned.

Note: 'baz' is required even if I don't want to type it
This does not work in TypeScript :
```
foo({ bar, baz }: { bar: string }) {
  //
}
```

**Comment by soda0289** [comment]
I was able to reproduce this parsing error with the following code:
```
class hello {
    foo({ bar, baz }: { bar: string, baz }) {

    }
}
```

The stacktrace:
```
TypeError: Cannot read property 'loc' of null
    at convertTypeAnnotation (/home/reyad/Workspace/typescript-eslint-parser/lib/ast-converter.js:498:32)
    at /home/reyad/Workspace/typescript-eslint-parser/lib/ast-converter.js:662:45
    at Array.forEach (native)
    at deeplyCopy (/home/reyad/Workspace/typescript-eslint-parser/lib/ast-converter.js:660:16)
    at convert (/home/reyad/Workspace/typescript-eslint-parser/lib/ast-converter.js:1974:17)
    at convertChild (/home/reyad/Workspace/typescript-eslint-parser/lib/ast-converter.js:485:20)
    at Array.map (native)
    at /home/reyad/Workspace/typescript-eslint-parser/lib/ast-converter.js:665:49
    at Array.forEach (native)
    at deeplyCopy (/home/reyad/Workspace/typescript-eslint-parser/lib/ast-converter.js:660:16)
```

**Comment by soda0289** [comment]
According to the typescript grammar the type is optional for interfaces and object members:
https://github.com/Microsoft/TypeScript/blob/master/doc/spec.md#391-property-signatures

Currently typescript parser assumes there is a type annotation when calling deeplyCopy(). Will submit PR with fix.

**Comment by soda0289** [comment]
Some possible bugs in typescript. I would think all the following examples are valid, at least according to the grammer:

Parses Correctly
```typescript
function foo({bar, baz}: {bar: string, baz}) {}
```
Parsing Fails
```typescript
function foo({bar, baz}: {bar, baz}) {}
```
Parses Correctly
```typescript
type foo = {bar: string, baz};
```
Parsing Fails
```typescript
type foo = {bar, baz};
```
Parses Correctly
```typescript
interface foo {
  bar;
  baz;
}

**Comment by soda0289** [comment]
The above comment is actually fixed in TypeScript 2.1 and all examples parse correctly.
